apps/mobile-wallet/src/persistent-storage/wallet.ts

```diff
@@ -87,18 +198,40 @@ export const generateAndStoreWallet = async (
   }
 }
 
-export const getWalletMetadata = async (): Promise<WalletMetadata | null> => {
-  const rawWalletMetadata = await getWithReportableError(WALLET_METADATA_STORAGE_KEY)
+const generateAndStoreWalletMetadata = async (name: WalletStoredState['name'], isMnemonicBackedUp: boolean) => {
+  const firstAddressHash = await generateAndStoreAddressKeypairForIndex(0)
+  const walletMetadata = generateWalletMetadata(name, firstAddressHash, isMnemonicBackedUp)
+  await storeWalletMetadata(walletMetadata)
+
+  return {
+    id: walletMetadata.id,
+    firstAddressHash
+  }
+}
+
+export const getWalletMetadata = async (throwError = true): Promise<WalletMetadata | null> => {
+  let rawWalletMetadata
+  let walletMetadata = null
 
   try {
-    return rawWalletMetadata ? JSON.parse(rawWalletMetadata) : null
+    rawWalletMetadata = await getWithReportableError(WALLET_METADATA_STORAGE_KEY)
   } catch (error) {
-    sendAnalytics({ type: 'error', error, message: 'Could not parse wallet metadata' })
-    throw error
+    if (throwError) throw error
+  }
+
+  if (rawWalletMetadata) {
+    try {
+      walletMetadata = JSON.parse(rawWalletMetadata)
+    } catch (error) {
+      sendAnalytics({ type: 'error', error, message: 'Could not parse wallet metadata' })
+      if (throwError) throw error
+    }
   }
+
+  return walletMetadata
 }
 
-export const getStoredWallet = async (error?: string): Promise<WalletMetadata> => {
+export const getStoredWalletMetadata = async (error?: string): Promise<WalletMetadata> => {
```

Comment:
We need to remove some of these function. It's confusing to me to have both `getStoredWalletMetadata` and `getWalletMetadata`, especially since the former rely directly on the latter.

Reply:
Added a TODO for now